src/main/solution_space/serialized_code.py

```diff
@@ -5,28 +5,101 @@
 import os
 import ast
 import logging
-from typing import List, Dict, Callable, Optional, Set
+from typing import List, Callable, Optional, Set
 
 from src.main.util import consts
 from src.main.util.consts import TASK
 from src.main.util.log_util import log_and_raise_error
 from src.main.canonicalization.consts import TREE_TYPE
-from src.main.util.file_util import create_file, is_file
 from src.main.util.helper_classes.id_counter import IdCounter
 from src.main.solution_space.data_classes import CodeInfo, User
 from src.main.util.language_util import get_extension_by_language
 from src.main.util.helper_classes.pretty_string import PrettyString
+from src.main.util.file_util import create_file, is_file, add_suffix_to_file
 from src.main.splitting.tasks_tests_handler import check_tasks, create_in_and_out_dict
-from src.main.canonicalization.canonicalization import are_asts_equal, get_code_from_tree, get_trees
+from src.main.canonicalization.canonicalization import are_asts_equal, get_code_from_tree, get_trees, \
+    get_nodes_number_in_ast
 
 log = logging.getLogger(consts.LOGGER_NAME)
 
 
-class AnonTree(IdCounter, PrettyString):
-    def __init__(self, anon_tree: ast.AST, code_info: Optional[CodeInfo] = None):
-        self._tree = anon_tree
+class ISerializedObject:
+    def __init__(self, folder_with_files: str, file_prefix: str,
+                 language: consts.LANGUAGE = consts.LANGUAGE.PYTHON):
+        self._folder_with_files = folder_with_files
+        self._file_prefix = file_prefix
+        self._language = language
+
+    @property
+    def folder_with_files(self) -> str:
+        return self._folder_with_files
+
+    @property
+    def file_prefix(self) -> str:
+        return self._file_prefix
+
+    @property
+    def language(self) -> consts.LANGUAGE:
+        return self._language
+
+    def get_file_path(self, suffix: Optional[str] = None, object_id: Optional[int] = None) -> str:
+        extension = get_extension_by_language(self._language)
+        file_name = f'{self._file_prefix}'
+        if object_id is not None:
+            file_name += f'_{str(object_id)}'
+        if suffix is not None:
+            file_name += f'_{suffix}'
+        return os.path.join(self._folder_with_files, f'{file_name}{str(extension.value)}')
+
+
+class SerializedTree:
+    def __init__(self, file_path: str, tree: ast.AST, tree_id: int, to_create_file: bool = True):
+        self._file_path = add_suffix_to_file(file_path, str(tree_id))
+        self._tree = tree
+        self._tree_file = None
```

Comment:
What is the difference between file_path and tree_file? Maybe we can remove one of them?